### DISCUSSION TITLE
Feature Requests and Roadmap

### DISCUSSION THREAD

**Discussion by haxzie** [haxzie/snipp.in#53] [Ideas]
There are a couple of huge features I'm planning to build. This includes the below-mentioned few items.

1. Draw/Whiteboard files
   - Users can create a `.draw` file to start drawing flowcharts or use it as a scratchpad directly within snipp
   - There files can then be downloaded as images or source files (JSON or SVG depending upon the implementation)
2. Cloud Storage and Accounts
   - To enable cloud sync, we might need to create login and signup to snipp
   - This might come with other problems such as loading time for files since snipp used to store everything locally this might be an issue. Can think about a way we can use a local first strategy but this will limit us a lot when adding new features like collaboration
3. Public Files and Pages
   - We recently announced creating Documents within snipp. It's be cool to have a share button to make the file instantly available as a web page and be able to share the documents for public viewing.
   - This will only be possible once the users start creating accounts
4. Collaborative Workspaces
   - This is will enable workspaces within snipp with ability to collaborate with others in the team. 
   - We can include live editing mode in documents and editor to give much better experience.
   
Any other feedback or suggestions are always welcomed :pray:

**Comment by boehs** [comment]
I think local first is the right move. Compare hashes of each note, if hash is different ask server for file. Simple, pretty effective
